fabric-dimensions-v1/src/main/java/net/fabricmc/fabric/api/dimension/v1/FabricDimensionType.java

```diff
@@ -0,0 +1,246 @@
+	 * @param defaultPlacer a default {@code EntityPlacer} for the dimension
+	 * @param hasSkyLight   {@code true} if the dimension type should have skylight like the overworld,
+	 *                      {@code false} otherwise
+	 * @see #builder()
+	 */
+	private FabricDimensionType(String suffix, String saveDir, Builder builder) {
+//			BiFunction<World, DimensionType, ? extends Dimension> factory, EntityPlacer defaultPlacer, boolean hasSkyLight) {
+		// Pass an arbitrary raw id that does not map to any vanilla dimension. That id should never get used.
+		super(3, suffix, saveDir, builder.factory, builder.skyLight, builder.biomeFunction);
+		this.defaultPlacement =builder.defaultPlacer;
+	}
+
+	/**
+	 * Return the desired raw id of this dimension type.
+	 *
+	 * @return the preferred raw id of this dimension type
+	 * @see Builder#desiredRawId(int)
+	 */
+	public int getDesiredRawId() {
+		return desiredRawId;
+	}
+
+	/**
+	 * Return the current raw id for this dimension type.
+	 *
+	 * <p> The returned id is guaranteed to be unique and persistent in a save,
+	 * as well as synchronized between a server and its connected clients.
+	 * It may change when connecting to a different server or opening a new save.
+	 *
+	 * @return the current raw id for this dimension type
+	 * @see #getDesiredRawId()
+	 */
+	@Override
+	public int getRawId() {
+		return this.fixedRawId;
+	}
+
+	/**
+	 * Return the default placement logic for this dimension. The returned placer
+	 * never returns {@code null} when called.
+	 *
+	 * @return the default placement logic for this dimension
+	 * @see FabricDimensions#teleport(Entity, DimensionType, EntityPlacer)
+	 */
+	public EntityPlacer getDefaultPlacement() {
+		return this.defaultPlacement;
+	}
+
+	/**
+	 * A builder for creating and registering {@code FabricDimensionType} instances. Example: <pre>   {@code
+	 *
+	 *   public static final FabricDimensionType MY_DIMENSION
+	 *       = FabricDimensionType.builder()
+	 *           .defaultPlacement((oldEntity, destination, portalDir, horizontalOffset, verticalOffset) ->
+	 *           		new BlockPattern.TeleportTarget(new Vec3d(0, 100, 0), teleported.getVelocity(), 0))
+	 *           .factory(MyDimension::new)
+	 *           .skyLight(true)
+	 *           .buildAndRegister();}</pre>
+	 *
+	 * <p>Builder instances can be reused; it is safe to call {@link #buildAndRegister(Identifier)} multiple
+	 * times (with different identifiers) to build and register multiple dimension types in series.
+	 * Each new dimension type uses the settings of the builder at the time it is built.
+	 *
+	 * @see FabricDimensionType#builder()
+	 */
+	public static final class Builder {
+		private EntityPlacer defaultPlacer;
+		private BiFunction<World, DimensionType, ? extends Dimension> factory;
+		private int desiredRawId = 0;
+		private boolean skyLight = true;
+		private class_4545 biomeFunction = class_4547.INSTANCE;
```

Comment:
For this field's name, I suggest `biomeAccessStrategy` as reasoned in https://github.com/FabricMC/yarn/pull/862. 
I also suggest we delegate to 4546 by default (as in 1.15+ we probably want 3d biomes over 2d ones)